.vscode/extensions.json

```diff
@@ -0,0 +1,8 @@
+{
+    "recommendations": [
+        "ms-vscode.cpptools",
+        "ms-vscode.cmake-tools",
+        "marus25.cortex-debug",
+        "twxs.cmake"
```

Comment:
I just noticed this is a different CMake extension than I though. Are there any main differences between this and https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools ? Is it possible that using the Microsoft extension we could not need to edit build.py and CMake files?

Reply:
I've just tested and the base CMake extension isn't actually required for builds; it's just there for giving syntax highlights etc. So we could remove both for this PR, and then place cmake-tools into another commit in the other PR?

Reply:
Yeah. I'd be specially interested to know if cmake-tools had the same problems with the `CMAKE_C_COMPILER` variable.

Reply:
Done 24ff078 and 2d79649.

> I'd be specially interested to know if cmake-tools had the same problems with the CMAKE_C_COMPILER variable.

I'd imagine so considering the CMake extension never provided build compat.